### PR TITLE
Generate separate json files per test.

### LINKED ISSUE #28: Add test id to json.
Currently web part has to combine id itself and it's not very convenient, i.e. `"id": "$package_name$class_name$name"`.

### LINKED ISSUE #29: Generate a separate json file for each test.
And minimize info for each test in `suite.json`.

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
Closes #29, closes #28.

### CODE REVIEW

File: composer/src/main/kotlin/com/gojuno/composer/html/HtmlFullSuite.kt [hunk 1]
```diff
@@ -11,7 +11,7 @@ data class HtmlFullSuite(
         val id: String,
         
         @SerializedName("tests")
-        val tests: List<HtmlTest>,
+        val tests: List<HtmlShortTest>,
```
Comment (ming13) [inline composer/src/main/kotlin/com/gojuno/composer/html/HtmlFullSuite.kt]: * `HtmlFullTest` → `HtmlTest`
* `HtmlShortTest` → `HtmlTestSummary`

?

File: composer/src/main/kotlin/com/gojuno/composer/html/HtmlFullSuite.kt [hunk 2]
```diff
@@ -31,7 +31,7 @@ data class HtmlFullSuite(
 
 fun Suite.toHtmlFullSuite(id: String) = HtmlFullSuite(
         id = id,
-        tests = tests.map { it.toHtmlTest() },
+        tests = tests.map { it.toHtmlFullTest().toHtmlShortTest() },
```
Comment (ming13) [inline composer/src/main/kotlin/com/gojuno/composer/html/HtmlFullSuite.kt]: Well, this is ugly.

File: composer/src/test/kotlin/com/gojuno/composer/html/HtmlReportSpec.kt [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+import org.jetbrains.spek.api.dsl.context
+import org.jetbrains.spek.api.dsl.it
+import rx.observers.TestSubscriber
+import java.io.File
+import java.util.concurrent.TimeUnit.MILLISECONDS
+import java.util.concurrent.TimeUnit.SECONDS
+
+class HtmlReportSpec : Spek({
+
+    context("writeHtmlReport") {
```
Comment (ming13) [inline composer/src/test/kotlin/com/gojuno/composer/html/HtmlReportSpec.kt]: Kind of useless declaration.

File: composer/src/test/kotlin/com/gojuno/composer/html/HtmlReportSpec.kt [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+                        ),
+                        passedCount = 2,
+                        ignoredCount = 0,
+                        failedCount = 1,
+                        durationNanos = MILLISECONDS.toNanos(1234 * 2),
+                        timestampMillis = 1805
+                )
+        )
+
+        val outputDir by memoized { File("${System.nanoTime()}") }
```
Comment (ming13) [inline composer/src/test/kotlin/com/gojuno/composer/html/HtmlReportSpec.kt]: `TemporaryFolder` can be used for that.

File: composer/src/main/kotlin/com/gojuno/composer/html/HtmlShortTest.kt [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        val status: HtmlFullTest.Status,
+        
+        @SerializedName("deviceId")
+        val deviceId: String,
+
+        @SerializedName("properties")
+        val properties: Map<String, Any>
+)
+
+
```
Comment (yunikkk) [inline composer/src/main/kotlin/com/gojuno/composer/html/HtmlShortTest.kt]: Extra line?

File: composer/src/main/kotlin/com/gojuno/composer/html/HtmlFullSuite.kt (hunk 1 above)
Comment (artem-zinnatullin) [inline composer/src/main/kotlin/com/gojuno/composer/html/HtmlFullSuite.kt]: niet

File: composer/src/main/kotlin/com/gojuno/composer/html/HtmlFullSuite.kt (hunk 2 above)
Comment (artem-zinnatullin) [inline composer/src/main/kotlin/com/gojuno/composer/html/HtmlFullSuite.kt]: well it is :)

File: composer/src/test/kotlin/com/gojuno/composer/html/HtmlReportSpec.kt (hunk 3 above)
Comment (artem-zinnatullin) [inline composer/src/test/kotlin/com/gojuno/composer/html/HtmlReportSpec.kt]: nah

File: composer/src/test/kotlin/com/gojuno/composer/html/HtmlReportSpec.kt (hunk 4 above)
Comment (artem-zinnatullin) [inline composer/src/test/kotlin/com/gojuno/composer/html/HtmlReportSpec.kt]: Does it remove inner content?

File: composer/src/main/kotlin/com/gojuno/composer/html/HtmlShortTest.kt (hunk 5 above)
Comment (artem-zinnatullin) [inline composer/src/main/kotlin/com/gojuno/composer/html/HtmlShortTest.kt]: My bad thx

File: composer/src/test/kotlin/com/gojuno/composer/html/HtmlReportSpec.kt (hunk 4 above)
Comment (ming13) [inline composer/src/test/kotlin/com/gojuno/composer/html/HtmlReportSpec.kt]: Yes.

Comment (artem-zinnatullin) [inline composer/src/test/kotlin/com/gojuno/composer/html/HtmlReportSpec.kt]: You mean one from AssertJ?

**Pull request merged**